### ISSUE TITLE
[Discussion] Single Player with Checkpoints?

### ISSUE THREAD

**Issue by Tormentor667** [Realm667/WolfenDoom#1124]
After @Talon1024 has implemented the Checkpoint System, and after I have added and tested them in Single Player, I kinda liked the approach somehow, so I wanted to discuss this in general with you.

What about adding the option to the game, that Single Player games can be played with Respawning and Checkpoints instead of Loading and Saving? Maybe we could add this choice before or after the Chapter choice?

**Comment by Talon1024** [comment]
My first thoughts about this are to give the player a choice between reloading their saved game and respawning at the latest checkpoint, which could be very helpful if the player ended up saving at a bad time or place.

**Comment by Tormentor667** [comment]
Indeed, so one could consider making the checkpoints active by default?

**Comment by x3003** [comment]
Currently the player can't see which checkpoint he/she has reached. Could this be added ?

**Comment by Tormentor667** [comment]
It already has been added by @Talon1024 afaik

**Comment by x3003** [comment]
Well, the checkpoint gfx have been added, but when playing coop you should be able to tell the other[s] which checkpoint (number) you've passed, so they could catch up or wait for you.

**Comment by Tormentor667** [comment]
Good suggestion, I think this should be doable with a HudMessageBold

**Comment by Tormentor667** [comment]
No idea how this could be realised as people do not know "where" a checkpoint actually is. You could at least just die :D

**Comment by Tormentor667** [comment]
Reactivating this: 

**I'd suggest activating the checkpoints in general so the player has both options.** What do you think @Talon1024 @Ozymandias81 @AFADoomer @Username-N00b-is-not-available @DoomJedi @doomjuan ?

**Comment by MuadDib90** [comment]
Can I join?

What Talon said is very reasonable. 
> My first thoughts about this are to give the player a choice between reloading their saved game and respawning at the latest checkpoint, which could be very helpful if the player ended up saving at a bad time or place.

So the best thing could be:
Making an option to choose what to load by default after choosing a dificulty of an episode is simple and easy to access. Also I'd add an option to the game menu options to be able to change it in-game, if needed as Talon said.

**Comment by Tormentor667** [comment]
The question for me is: Do we really need an option? I mean, if "checkpoints" are active by default and the player respawns at an checkpoint after dying, he can still decide to load his savegame, as the autosave states are active as well.

**Comment by Tormentor667** [comment]
Reviving this: What is currently the state of the checkpoint system for both multiplayer and singleplayer?

@AFADoomer @Username-N00b-is-not-available @Talon1024
